### PR TITLE
dev/core#2039 Remove extraneous location queries from contact.create->Location::block path

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Removes 10 queries from each contact  create that are otherwise dealt with elsewhere
Updated version of https://github.com/civicrm/civicrm-core/pull/18477 

Before
----------------------------------------
After creating location blocks 2 queries are done for each of the 5 entities to ensure is_primary fields have not been messed up (ie contact has 1 or more phones but none are marked primary or contact has more than one primary phone).

After
----------------------------------------
Queries are removed as location entities are already created correctly

Technical Details
----------------------------------------
This PR is part of a series of PRs to make this possible - best documented at https://lab.civicrm.org/dev/core/-/issues/2039#note_47758 

Comments
----------------------------------------
https://lab.civicrm.org/dev/core/-/issues/2039

Note test cover added by adding a postAssert to ALL tests. This picked up multiple issues which I fixed in preparation for this

tldr is - I fixed this code to call create & delete functions that manage is_primary correctly (which I had to fix some of them to do) and no it is redundant in here

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton I think this may need a rebase

**Comment by eileenmcnaughton** [comment]
@seamuslee001 done

**Comment by colemanw** [comment]
Nice! Agree this has good test cover.
